locals.tf

```diff
@@ -9,37 +9,39 @@ locals {
 }
 
 locals {
-  locations_cached_or_live        = data.local_file.locations.content
-  regions_by_display_name         = { for v in local.regions_recommended_or_not : v.display_name => v }
-  regions_by_name                 = { for v in local.regions_recommended_or_not : v.name => v }
-  regions_by_name_or_display_name = merge(local.regions_by_display_name, local.regions_by_name)
-  regions_data_merged = [
-    for v in jsondecode(local.locations_cached_or_live).value :
-    merge(
-      {
-        name               = v.name
-        display_name       = v.displayName
-        paired_region_name = try(one(v.metadata.pairedRegion).name, null)
-        geography          = v.metadata.geography
-        geography_group    = v.metadata.geographyGroup
-        recommended        = v.metadata.regionCategory == "Recommended"
-      },
-      {
-        zones = sort(lookup(local.regions_to_zones_map, v.displayName, []))
-      }
-    ) if v.metadata.regionType == "Physical"
+  default_node_pool_zones = setsubtract(local.zones, local.restrictions)
```

Comment:
```suggestion
  default_node_pool_available_zones = setsubtract(local.zones, local.restrictions)
```

Reply:
also it might be better to have this at the end  because it is the last part that returns the zones and make it similar with the `filtered_vms_by_pool` local